daffodil-test/src/test/java/org/goodudfs/example/StringFunctions/Compare.java

```diff
@@ -0,0 +1,19 @@
+package org.goodudfs.example.StringFunctions;
+
+import java.io.Serializable;
+
+import org.apache.daffodil.udf.FunctionClassInfo;
+
+@FunctionClassInfo(
+		name = "compare",
+		namespace = "http://goodudfs.StringFunctions"
+)
+public class Compare implements Serializable {
```

Comment:
Is it possible to insist that all UDFs implement a common UDF interface? Something along the lines of:

interface UDF{
   public Object evaluate(Object ... args);
}

I'm not sure how variance works in Java, so it might not be possible to write this signature in such a way that actually allows reasonable signatures in the subclasses.

Even just interface UDF{} might be useful as a way for users to mark the fact that these classes are implementing UDFs.

Reply:
Makes sense to me. I do like the idea of enforcing that a UDF impelments a UserDefineFunction interface, even if it's empty. It would likely mean we could have places in code where the type is UserDefinedFunction rather than Object. Functionally does mean much, but probably makes the code clear and more readable. And it gives us a place to hang required/abstract functions if we decide they are needed.